src/libs/actions/Session/index.ts

```diff
@@ -81,22 +82,106 @@ Onyx.connect({
     callback: (val) => (preferredLocale = val),
 });
 
+function isSupportalToken(): boolean {
+    return sessionAuthTokenType === 'supportal';
+}
+
+/**
+ * Sets the SupportToken. This method will only be used on SignOut when an user
+ * is using a token of type support or on dev.
+ */
+function setSupportAuthToken(supportAuthToken: string, email?: string, accountID?: number) {
+    if (supportAuthToken) {
+        Onyx.merge(ONYXKEYS.SESSION, {
+            authTokenType: 'support',
+            authToken: supportAuthToken,
+            email,
+            accountID,
+        }).then(() => {
+            Log.info('[Supportal] Authtoken set');
+        });
+    } else {
+        Onyx.set(ONYXKEYS.SESSION, {}).then(() => {
```

Comment:
Changed to null in another function.

> don't we normally stash the old login and restore it when someone's done with supportal?

We won't implement this for new dot for now, but I think we could create a way to do that in the future. Like, if you're already logged, we clear everything and store the credentials so when you go back to your stashed login we set those as the current session.

Reply:
FWIW, I feel like that could be quite easy to include in the V1 and would save internal employees a good amount of time.

Reply:
Basically, just:

1. Create a `STASHED_SESSION` Onyx key
1. When support logging in, take the existing `SESSION`, if we have one, and move it from `SESSION` to `STASHED_SESSION`
1. When support logging out, take the existing `SESSION`, replace it with `STASHED_SESSION`, and then delete the `STASHED_SESSION`

Seems pretty easy to do with relatively few lines of changes. Not going to block if you feel strongly that it's not worth it, but also happy to provide guidance if it would encourage you to implement it.

Reply:
We would need to add a `restore stashed session` somewhere inside the app. Clicking on `Restore stashed login` on supportal wouldn't do that automatically on new dot.

Where would you suggest we add that?

Reply:
I would suggest that if you're support logged in the normal Sign Out button is replaced by Restore Stashed Login. Nice and simple